gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/shift/SweIdccShiftDispatcher.java

```diff
@@ -32,24 +32,24 @@ public Map<String, Double> dispatch(double stepExchangeValue) {
         Map<String, Double> factors = new TreeMap<>();
         switch (direction) {
             case ES_FR:
-                factors.put(toEic("PT"), initialNetPositions.get(toEic("PT")));
-                factors.put(toEic("ES"), -initialNetPositions.get(toEic("PT")) + stepExchangeValue);
-                factors.put(toEic("FR"), -stepExchangeValue);
+                factors.put(toEic("PT"), 0.);
+                factors.put(toEic("ES"), -initialNetPositions.get(toEic("PT")) - initialNetPositions.get(toEic("ES")) + stepExchangeValue);
+                factors.put(toEic("FR"), -(factors.get(toEic("PT")) + factors.get(toEic("ES"))));
```

Comment:
You need to substract the initial NP of France here also. Or to simplify the formula factors.put(toEic("FR"), -exchangeValue - - initialNetPositions.get(toEic("FR")). Do you agree with that proposal ?